From 58695fbf11280b3b569fb94c0f669b112eb21ce9 Mon Sep 17 00:00:00 2001
From: Tim Hartung <thart@math.uni-paderborn.de>
Date: Thu, 10 Nov 2011 18:40:31 +0100
Subject: [PATCH] dpkg-maintscript-helper.sh: Fix wrong comparsion

Due to a wrong operator supplied to dpkg --compare-versions
dpkg-maintscript-helper.sh did not remove obsolete conffiles in
some cases.
---
 scripts/dpkg-maintscript-helper.sh |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/scripts/dpkg-maintscript-helper.sh b/scripts/dpkg-maintscript-helper.sh
index 631a5a1..d3f6e3e 100755
--- a/scripts/dpkg-maintscript-helper.sh
+++ b/scripts/dpkg-maintscript-helper.sh
@@ -54,13 +54,13 @@ rm_conffile() {
 	case "$DPKG_MAINTSCRIPT_NAME" in
 	preinst)
 		if [ "$1" = "install" -o "$1" = "upgrade" ] && [ -n "$2" ] &&
-		   dpkg --compare-versions "$2" le-nl "$LASTVERSION"; then
+		   dpkg --compare-versions "$2" gt "$LASTVERSION"; then
 			prepare_rm_conffile "$CONFFILE" "$PACKAGE"
 		fi
 		;;
 	postinst)
 		if [ "$1" = "configure" ] && [ -n "$2" ] &&
-		   dpkg --compare-versions "$2" le-nl "$LASTVERSION"; then
+		   dpkg --compare-versions "$2" gt "$LASTVERSION"; then
 			finish_rm_conffile $CONFFILE
 		fi
 		;;
@@ -71,7 +71,7 @@ rm_conffile() {
 		fi
 		if [ "$1" = "abort-install" -o "$1" = "abort-upgrade" ] &&
 		   [ -n "$2" ] &&
-		   dpkg --compare-versions "$2" le-nl "$LASTVERSION"; then
+		   dpkg --compare-versions "$2" gt "$LASTVERSION"; then
 			abort_rm_conffile "$CONFFILE"
 		fi
 		;;
-- 
1.7.1

